firefox-ios/Client/Frontend/Library/Bookmarks/Edit Bookmark/EditBookmarkViewController.swift

```diff
@@ -77,7 +77,17 @@ class EditBookmarkViewController: UIViewController,
         super.viewDidLoad()
         title = .Bookmarks.Menu.EditBookmarkTitle
         viewModel.onFolderStatusUpdate = { [weak self] in
-            self?.tableView.reloadSections(IndexSet(integer: Section.folder.rawValue), with: .automatic)
+            guard let self = self else { return }
+
+            // TODO: FXIOS-10985 - replace batch updates with diffable data source
+            self.tableView.performBatchUpdates {
+                if self.viewModel.isFolderCollapsed {
+                    self.tableView.deleteSections(IndexSet(integer: Section.newFolder.rawValue), with: .none)
+                } else {
+                    self.tableView.insertSections(IndexSet(integer: Section.newFolder.rawValue), with: .none)
+                }
+                self.tableView.reloadSections(IndexSet(integer: Section.folder.rawValue), with: .automatic)
+            }
```

Comment:
should we remove this as well since we're not planning on using this?

Reply:
This is required to hide/show the "new folder" cell. Are you suggesting to close the PR in favor of the diffable data source ticket?

Reply:
I was thinking that there would be a part I and II where we can merge the other code and then work on this piece with the diffable data source, but I'm okay with approving this as well since it's behind the `isBookmarkRefactorEnabled` feature flag.